src/app/ReadClient.cpp

```diff
@@ -957,34 +972,34 @@ void ReadClient::OnResubscribeTimerCallback(System::Layer * apSystemLayer, void
     _this->mNumRetries++;
 }
 
-bool ReadClient::ResubscribeIfNeeded()
+void ReadClient::ResubscribeIfNeeded(bool & aResubscribe, uint32_t & aNextResubscribeIntervalMsec)
```

Comment:
Why wouldn't we leave the boolean as the return value?  then the code in this method would stay pretty much as-is, the caller would ignore the outparam if `false` is returned (which is normal for how outparams work), and this method would have much clearer behavior...

Reply:
updated with your suggestion, thanks